gf-webapp/src/component/filter/SearchFilter.tsx

```diff
@@ -0,0 +1,45 @@
+import {useEffect} from "react";
+import {useDictionary} from "src/logic/DictionaryContext/useDictionary";
+import {useFilterRooms} from "src/component/filter/FilterContext";
+import {useSearchParams} from "react-router-dom";
+import {FilterInput, URL_QUERY_KEYS, InputType} from "./FilterInput";
+
+
+export const SearchFilter = () => {
+  const {bookingRoomForm} = useDictionary().dictionary;
+  const [params, setParams] = useSearchParams();
+  const {searchValue, setSearchValue} = useFilterRooms();
+
+  useEffect(() => {
+    if (params.has(URL_QUERY_KEYS.Search)) {
+      setSearchValue(params.get(URL_QUERY_KEYS.Search) as string);
```

Comment:
Avoid as

Reply:
For don't use as converted the next lines into
    const paramValueRaw = params.get(props.urlQueryKey);
      const paramValueNumber: number = paramValueRaw ? JSON.parse(paramValueRaw) : 0;
      const paramValueString: string = paramValueRaw ? paramValueRaw.slice(1, -1) : "";
    But i think it's not the best variant... Please help)